handler.py

```diff
@@ -54,48 +57,59 @@ def fxa_validate(event, device_id=None):
     """
     # extract the FxA OAuth token from the Authorization header
     try:
-        auth = event["headers"]["authorization"]
+        assert ("bearer " in event["headers"]["authorization"].lower())
+        auth = event["headers"]["authorization"].split(' ')[1]
+
     except KeyError:
         raise HandlerException(
             status_code=401,
             message="Missing authorization header")
+    except IndexError:
+        raise HandlerException(
+            status_code=401,
+            message="Invalid authorization header"
+        )
     try:
-        response = urllib2.urlopen(
-            "https://oauth.accounts.firefox.com/v1/verify",
-            data=auth,
-            timeout=5).read()
-        token = json.loads(response)["token"]
-        if token == "https://identity.mozilla.com/apps/pushbox/":
-            return ["send", "recv"]
-        elif token == "https://identity.mozilla.com/apps/pushbox/send":
-            return ["send"]
-        elif token == "https://identity.mozilla.com/apps/pushbox/recv":
-            return ["recv"]
-        elif (token ==
-              "https://identity.mozilla.com/apps/pushbox/send/{}".format(
-                  device_id)):
-            return ["send"]
-        elif (token ==
-              "https://identity.mozilla.com/apps/pushbox/recv/{}".format(
-                  device_id)):
-            return ["recv"]
+        req = request.Request(
+            "https://{}/v1/verify".format(FXA_HOST),
+            method="POST",
+            data=json.dumps({"token": auth}).encode('utf8'),
+            headers={"Content-type": "application/json"})
+        response = request.urlopen(req, timeout=5).read()
+        scopes = json.loads(response)["scope"]
```

Comment:
You could consider `set(json.loads(response)["scope"])` here, because you do a bunch of `in` tests on it below.  It'll *probably* be a short list of scopes that's cheap it iterate over, but nothing in the system guarantees that.